client/containers/Pub/PubDocument/Review/ReviewEditor.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from 'react';
+
+import { MinimalEditor } from 'components';
+import { DocJson } from 'types';
+import { useLocalStorage } from 'client/utils/useLocalStorage';
+import { getEmptyDoc } from 'client/components/Editor';
+
+require('./reviewEditor.scss');
+
+type Props = {
+	setReviewDoc: any;
+	communityData: any;
+	pubData: any;
+};
+
+const ReviewEditor = (props: Props) => {
+	const {
+		setReviewDoc,
+		communityData,
+		pubData: { id },
+	} = props;
+
+	const { value: review, setValue: setReview } = useLocalStorage<DocJson>({
+		initial: () => getEmptyDoc(),
+		communityId: communityData.id,
+		featureName: 'new-review-editor',
+		path: [`pub-${id}`],
+		debounce: 100,
+	});
+
+	setReviewDoc(review);
+
+	const updatingReviewDoc = (doc: DocJson) => {
+		setReview(doc);
+		setReviewDoc(review);
+	};
+
+	useEffect(() => {
+		setReviewDoc(review);
+	}, [review, setReviewDoc]);
```

Comment:
And now that I think about it, if this is in the parent `<Review>` component, it won't even need to `useState`, since that state management is handled in `useLocalStorage`

Reply:
yeah, but i have to call the localStorage within `ClientOnly`. and i would rather not wrap the entire review component

Reply:
Hmm, that's annoying. Maybe I should add a server-side fallback to `useLocalStorage()` instead of forcing consumers to do that.